### ISSUE TITLE
The new "hidden loader internals" PR can cause classloading issues

### ISSUE THREAD

**Issue by AlexIIL** [QuiltMC/quilt-loader#221]
For example:
```java
java.lang.ClassCastException class net.ludocrypt.corners.world.chunk.YearningCanalChunkGenerator
cannot be cast to class net.minecraft.class_3754
<snip>.YearningCanalChunkGenerator is in unnamed module of loader <snip>.KnotClassLoader$Separate @7f4b9586; 
net.minecraft.class_3754 is in unnamed module of loader <snip>.KnotClassLoader$Separate @7799d031)
	at net.minecraft.server.MinecraftServer.handler$ebe004$onInit(MinecraftServer.java:2790)
	at net.minecraft.server.MinecraftServer.<init>(MinecraftServer.java:295)
```
This is not a mod issue, it just means we can't use separate classloaders to implement mod-to-mod hidden internals - I'll migrate to some ASM in the transformer cache.

**Comment by AlexIIL** [comment]
It turns out this can occur on 0.17.10 as well, so this specific crash might not be a loader problem. I still want to switch to using some ASM after the transform cache though.

Full crash report: [crash-2023-01-25_16.49.56-client.txt](https://github.com/QuiltMC/quilt-loader/files/10515019/crash-2023-01-25_16.49.56-client.txt)

**Comment by TheGlitch76** [comment]
I believe this was resolved with f3cf272fddaecd9b9c8376dce94fe3e0df0ce551
